### DISCUSSION TITLE
Put all umbraco data types into a folder under Data Types

### DISCUSSION THREAD

**Discussion by seanrockster** [umbraco/Umbraco-CMS#13054] [Features and ideas]
e.g. Data Types > Umbraco (all umbraco base data types)

We organise our data types into subfolders however the standard data types are all under the root. Not a major issue however when creating new types via document types, they get dropped into the root. If the standard types are segregated into their own folder its easier to spot the ones we've created.

Alternatively, have a way to move multiple types, because moving them one by one is painful.

**Comment by skttl** [comment]
You could write a notification handler for DataTypeSaving where you can move the data type into the the appropriate folder if it isn't where you want it.

https://our.umbraco.com/documentation/Reference/Notifications/DataTypeService-Notifications/

**Reply by seanrockster** [reply]
Appropriate folder is subjective, the handler would have to know which type goes where and that could change over time, meaning we'd need to update the handler. But could be used to at least put it into the root of our custom types folder.

**Reply by skttl** [reply]
> Appropriate folder is subjective, the handler would have to know which type goes where and that could change over time, meaning we'd need to update the handler.

Yeah, but at least you'll get to have your own opinion on how they are sorted

**Reply by seanrockster** [reply]
true

**Comment by Bakersbakebread** [comment]
I love this idea, actually. The data type folder can become a somewhat untamed beast if you neglect it for just a second. 

Even having them 'grouped' by their base implementation would make them more manageable 

![image](https://user-images.githubusercontent.com/29239704/191982927-5a406d5a-3d0e-4679-a94a-30f5892fa992.png)

**Reply by simonbusborg** [reply]
Grouped and separated from newly created, like all predefined data types are put in a folder with in a `Default` folder.

**Comment by LottePitcher** [comment]
And I still remember when you couldn't use folders...!  

A folder called "Custom" and then sub folders by type is an implementation I've seen and used myself a lot. But I've also seen a folder called the name of the project instead of "Custom".  It's definitely subjective.

So just thinking about why I like to put in a 'custom' folder... Two reasons I think:
1 - It helps me know what I've added for this site, rather than what came OOTB with Umbraco. 
2 - I never change the core/default data types. If I want to change them I add a new one.

If you've ever inherited a site and all the data types are in the root then your heart sinks, it takes more mental effort to work out what has been added for this site. 

So when I expand the data types folder to find the one I want to edit, or the place to add one, it's never in the root! It would be much better to move those core ones 'out the way'! 

So I think @simonbusborg 's and @seanrockster suggestion of having those predefined/core/default datatypes created in their own folder makes a lot of sense and gets my 👍   And has the benefit that data types created whilst working on doc types being put in the root then isn't a problem. Then people can implement whatever folder structure they like, or just create all their data types in the root.

Unless of course this makes it discoverability of data type configuration harder for people new to Umbraco of course

**Comment by seanrockster** [comment]
Or better yet, when creating a new type, prompt the user to save the new type to a specific folder, otherwise it goes to the root.

**Comment by mistyn8** [comment]
Think this is also tied to the add property/datatype workflow, it's far too easy to create 10's of datatypes that should just reuse a single existing one with the current journey (esp when in an infinite editor scenario). Perhaps a confirmation prompt, are you sure you want to create this new datatype, or reuse one of the same type and list them?

Agree with @Bakersbakebread  Some clients like all datatypes of the same type to exist in a type folder, seems a relatively straight forward config option? Also means they see immediately the create vs reuse issue above!

Also lots of times we don't actually need all the OOTB dataypes for a project, install step/flag to don't add any? (similar to choosing /adding a starter kit). Infact could datatypes/mediatypes be repurposed to a starter kit?

**Reply by mistyn8** [reply]
Dataype merging to get rid of all the unnecessary dupes of toggle and update all doctypes using them?

**Reply by mistyn8** [reply]
Slightly of track.. mediatype folder, would be nice to move the inheritance into the concrete node.. so not having to create image-folder, file-folder, pdf-folder, slide-folder etc just to have allowed child items.

**Comment by hfloyd** [comment]
One of the first things I do in a new installation now is to create a "Standard" folder under DataTypes and move all the Umbraco default ones there😁
I usually don't put all my custom ones into folders; I name them starting with the Property Editor, so they sort alphabetically by "type" nicely in the root. 

I mostly want the Standard ones segregated so I don't edit or delete them.
